### PR TITLE
all constructors in React components should accept props

### PR DESCRIPTION
**Pull request by Justicea83**
### Fixed Issues
$https://github.com/Expensify/App/issues/7821

### Tests
1. Confirm all tests still pass

### CODE REVIEW

File: eslint-plugin-expensify/tests/prefer-constructors-with-props.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const message = require('../CONST').MESSAGE.PREFER_CONSTRUCTORS_WITH_PROPS;
+
+const ruleTester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module',
+    },
+});
+
+const goodExample = `
```
Comment (marcaaron) [inline eslint-plugin-expensify/tests/prefer-constructors-with-props.test.js:12]: Let's add a good exmple like:

```js
class Random {
    constructor() {
    }
}
```

File: eslint-plugin-expensify/CONST.js [hunk 2]
```diff
@@ -10,5 +10,6 @@ module.exports = {
         PREFER_IMPORT_MODULE_CONTENTS: 'Do not import individual exports from local modules. Prefer \'import * as\' syntax.',
         PREFER_ONYX_CONNECT_IN_LIBS: 'Only call Onyx.connect() from inside a /src/libs/** file. React components and non-library code should not use Onyx.connect()',
         PREFER_UNDERSCORE_METHOD: 'Prefer \'_.{{method}}\' over the native function.',
+        PREFER_CONSTRUCTORS_WITH_PROPS: 'Constructor should accept props.',
```
Comment (marcaaron) [inline eslint-plugin-expensify/CONST.js:13]: This should be more specific I think..
```suggestion
        PREFER_REACT_CLASS_CONSTRUCTORS_WITH_PROPS: 'React class constructors must accept props.',
```

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: eslint-plugin-expensify/tests/prefer-constructors-with-props.test.js (hunk 1 above)
Comment (Justicea83) [reply eslint-plugin-expensify/tests/prefer-constructors-with-props.test.js:12]: sure, i'll add it and am sure it's gonna fail.

**Comment by Justicea83** [comment]
Closed it because I realized the branch name was wrong.

**Pull request closed without merging**